client/java/generator/src/main/java/io/openlineage/client/Generator.java

```diff
@@ -16,67 +24,155 @@
+    if (schema.has("$id") && schema.get("$id").isTextual()) {
+      URL idURL = new URL(schema.get("$id").asText());
+      try (InputStream openStream = idURL.openStream();) {
+        JsonNode published = mapper.readValue(openStream, JsonNode.class);
+        if (!published.equals(schema)) {
+          throw new RuntimeException("You must increment the version when modifying the schema. The current schema at " + url + " has the $id " + idURL + " but the version at that URL does not match.");
+        }
+      } catch (FileNotFoundException e) {
+        logger.warn("This version of the spec is not published yet: " + idURL);
+      }
+      return idURL;
     }
+    return url;
   }
 
-  public static void generate(String baseURL, InputStream input, File output) {
-    if (!output.getParentFile().exists()) {
-      if (!output.getParentFile().mkdirs()) {
-        throw new RuntimeException("can't create output " + output.getAbsolutePath());
+  public static void generate(Set<URL> urls, File outputBase) {
+    if (!outputBase.exists()) {
+      if (!outputBase.mkdirs()) {
+        throw new RuntimeException("can't create output " + outputBase.getAbsolutePath());
       }
     }
     try {
-      ObjectMapper mapper = new ObjectMapper();
-      JsonNode schema = mapper.readValue(input, JsonNode.class);
-      if (schema.has("$id") && schema.get("$id").isTextual()) {
-        String idURL = schema.get("$id").asText();
-        try (InputStream openStream = new URL(idURL).openStream();) {
-          JsonNode published = mapper.readValue(openStream, JsonNode.class);
-          if (!published.equals(schema)) {
-            throw new RuntimeException("You must increment the version when modifying the schema. The current schema at " + baseURL + " has the $id " + idURL + " but the version at that URL does not match.");
-          }
-        } catch (FileNotFoundException e) {
-          logger.warn("This version of the spec is not published yet: " + e.toString());
-        } finally {
-          baseURL = idURL;
+      TypeResolver typeResolver = new TypeResolver(urls);
+
+      // We add the facets to the core model here to keep code generation convenient
+      Map<String, ObjectResolvedType> facetContainers = indexFacetContainersByType(typeResolver);
+      enrichFacetContainersWithFacets(typeResolver, facetContainers);
+
+      Map<String, URL> containerToID = new HashMap<>();
+      for (URL url : urls) {
+        String path = url.getPath();
+        url = verifySchemaVersion(url);
+        if (path.endsWith(JSON_EXT)) {
+          String containerClassName = path.substring(path.lastIndexOf('/') + 1, path.lastIndexOf('.'));
+          containerToID.put(containerClassName, url);
+        } else {
+          throw new IllegalArgumentException("inputs should end in " + JSON_EXT + " got " + path);
         }
       }
-      TypeResolver typeResolver = new TypeResolver(schema);
+
+      String containerClassName = "OpenLineage";
```

Comment:
_Minor:_ Can these be defined as class constants?

```java
CONTAINER_CLASS_NAME = "OpenLineage";
JAVA_PATH = CONTAINER_CLASS_NAME+ "OpenLineage";
```

Reply:
Ok